adapters/richaudience/richaudience.go

```diff
@@ -0,0 +1,299 @@
+package richaudience
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/mxmCherry/openrtb/v15/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/metrics"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+type richaudienceRequest struct {
+	ID     string             `json:"id,omitempty"`
+	Imp    []openrtb2.Imp     `json:"imp,omitempty"`
+	User   richaudienceUser   `json:"user,omitempty"`
+	Device richaudienceDevice `json:"device,omitempty"`
+	Site   richaudienceSite   `json:"site,omitempty"`
+	Test   int8               `json:"test,omitempty"`
+}
+
+type richaudienceUser struct {
+	BuyerUID string              `json:"buyeruid,omitempty"`
+	Ext      richaudienceUserExt `json:"ext,omitempty"`
+}
+
+type richaudienceUserExt struct {
+	Eids    []openrtb_ext.ExtUserEid `json:"eids,omitempty"`
+	Consent string                   `json:"consent,omitempty"`
+}
+
+type richaudienceDevice struct {
+	IP   string `json:"ip,omitempty"`
+	IPv6 string `json:"ipv6,omitempty"`
+	Lmt  int8   `json:"lmt,omitempty"`
+	DNT  int8   `json:"dnt,omitempty"`
+	UA   string `json:"ua,omitempty"`
+}
+type richaudienceSite struct {
+	Domain string `json:"domain,omitempty"`
+	Page   string `json:"page,omitempty"`
+}
+
+// Builder builds a new instance of the RichAudience adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters.Bidder, error) {
+
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	richaudienceRequest := richaudienceRequest{}
+
+	raiHeaders := http.Header{}
+
+	setHeaders(&raiHeaders)
+
+	richaudienceRequest.ID = request.ID
+
+	isUrlSecure := setSite(request, &richaudienceRequest)
+
+	resImps, err := setImp(request, &richaudienceRequest, isUrlSecure)
+
+	if err != nil {
```

Comment:
Nitpick: Consider removing the line break on line 73 before this err check group the err check with the err variable for improved readability. ie:

```
resImps, err := setImp(request, &richaudienceRequest, isUrlSecure)
if err != nil {
  return nil, []error{err}
}
```

Reply:
in next commit i have solved